cwms-data-api/src/main/java/cwms/cda/formatters/json/adapters/TimeSeriesProfileParserSerializer.java

```diff
@@ -0,0 +1,98 @@
+/*
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Hydrologic Engineering Center
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package cwms.cda.formatters.json.adapters;
+
+import static cwms.cda.api.timeseriesprofile.TimeSeriesProfileParserBase.COLUMNAR_TYPE;
+import static cwms.cda.api.timeseriesprofile.TimeSeriesProfileParserBase.INDEXED_TYPE;
+
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.jsontype.TypeSerializer;
+import cwms.cda.data.dto.timeseriesprofile.ParameterInfoColumnar;
+import cwms.cda.data.dto.timeseriesprofile.ParameterInfoIndexed;
+import cwms.cda.data.dto.timeseriesprofile.TimeSeriesProfileParser;
+import cwms.cda.data.dto.timeseriesprofile.TimeSeriesProfileParserColumnar;
+import cwms.cda.data.dto.timeseriesprofile.TimeSeriesProfileParserIndexed;
+import java.io.IOException;
+
+
+public class TimeSeriesProfileParserSerializer extends JsonSerializer<TimeSeriesProfileParser> {
+    @Override
+    public void serialize(TimeSeriesProfileParser value, JsonGenerator gen, SerializerProvider provider)
+            throws IOException {
+        String type = value.getClass().getAnnotation(JsonTypeName.class).value();
+        gen.writeStartObject();
+        gen.writeObjectField("type", type);
```

Comment:
If you call `serializers.defaultSerializeValue(value, gen);` shouldn't that make it so that you only have to add in the custom "type" and everything else will operate as expected?

Reply:
The easiest way is to use a new ObjectMapper that doesn't have your custom serializer registered and then append into the JsonNode.